crates/ra_proc_macro_srv/src/dylib.rs

```diff
@@ -0,0 +1,205 @@
+//! Handles dynamic library loading for proc macro
+
+use crate::{proc_macro::bridge, rustc_server::TokenStream};
+use std::path::Path;
+
+use goblin::{mach::Mach, Object};
+use libloading::Library;
+use ra_proc_macro::ProcMacroKind;
+
+static NEW_REGISTRAR_SYMBOL: &str = "_rustc_proc_macro_decls_";
+
+fn get_symbols_from_lib(file: &Path) -> Option<Vec<String>> {
+    let buffer = std::fs::read(file).ok()?;
+    let object = Object::parse(&buffer).ok()?;
```

Comment:
I wonder if this can reutrn an `io::Error`? io::Error includes InvalidData, and ingeneral "parsing" errors should be convertable to io errors